rust_src/src/buffers.rs

```diff
@@ -553,19 +545,7 @@ impl From<LispObject> for LispBufferOrName {
     }
 }
 
-impl From<LispObject> for Option<LispBufferOrName> {
-    fn from(v: LispObject) -> Option<LispBufferOrName> {
-        if v.is_nil() {
-            None
-        } else if v.is_string() {
-            Some(LispBufferOrName::Name(v))
-        } else if v.is_buffer() {
-            Some(LispBufferOrName::Buffer(v))
```

Comment:
This is a change in behavior. Now `or_error` is called before, this would return None.